blockly-core/core/ui/block_svg/block_svg_functional.js

```diff
@@ -51,7 +51,19 @@ Blockly.BlockSvgFunctional.prototype.renderDraw_ = function(iconWidth, inputRows
 
   this.blockClipRect_.setAttribute('d', this.svgPath_.getAttribute('d'));
 
-  var rect = this.svgPath_.getBoundingClientRect();
+  // getBBox doesn't work on blocks that arent visible in FF
+  if (!this.block_.isVisible()) {
+    return;
+  }
+
+  // isVisible can report true if the block is in the modal blockspace but
+  // the function editor isn't yet open
+  if (this.block_.blockSpace === Blockly.functionEditor.modalBlockSpace &&
```

Comment:
As you mentioned in the comment, I think it'd make sense to move this visibility check in to the block's `isVisible`. One thing worth trying to retain is only calling `isOpen()` if this block is in the modal editor (since it uses a .style check under the hood, wouldn't want to call it in most circumstances)